### PR TITLE
Address deprecation warning on set-output

### PR DESCRIPTION
**Pull request by alexevanczuk**
# Summary

This PR addresses the following deprecation warning I'm receiving in my build that uses this action:
```
Annotations
2 warnings
deploy
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

The docs (https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files) say to use:
```
echo "{name}={value}" >> $GITHUB_OUTPUT
```

### CODE REVIEW

**Comment by alexevanczuk** [comment]
@cvx  Let me know if you'd have time to review and merge this change. If not, I'll close this PR.

**Comment by cvx** [comment]
Thanks for the PR!

> The docs (https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files) say to use:
>
> > echo "{name}={value}" >> $GITHUB_OUTPUT

That would mean the updated lines should look like this, right?

```sh
echo "gem_version=$gem_version" >> $GITHUB_OUTPUT
```

File: entrypoint.sh [hunk 1]
```diff
@@ -13,14 +13,14 @@ chmod 0600 ~/.gem/credentials
 set -x
 
 gem_version=$(ruby -r rubygems -e "puts Gem::Specification::load('$(ls *.gemspec)').version")
-echo "::set-output name=gem_version::$gem_version"
+echo "name=gem_version::$gem_version" >> $GITHUB_OUTPUT
```
Comment (alexevanczuk) [inline entrypoint.sh]:
```suggestion
echo "gem_version=$gem_version" >> $GITHUB_OUTPUT
```

File: entrypoint.sh [hunk 2]
```diff
@@ -13,14 +13,14 @@ chmod 0600 ~/.gem/credentials
...
 
 if git fetch origin "refs/tags/v$gem_version" >/dev/null 2>&1
 then
   echo "Tag 'v$gem_version' already exists"
-  echo "::set-output name=new_version::false"
+  echo "name=new_version::false" >> $GITHUB_OUTPUT
```
Comment (alexevanczuk) [inline entrypoint.sh]:
```suggestion
  echo "new_version=false" >> $GITHUB_OUTPUT
```

File: entrypoint.sh [hunk 3]
```diff
@@ -13,14 +13,14 @@ chmod 0600 ~/.gem/credentials
...
 else
-  echo "::set-output name=new_version::true"
+  echo "name=new_version::true" >> $GITHUB_OUTPUT
```
Comment (alexevanczuk) [inline entrypoint.sh]:
```suggestion
  echo "new_version=true" >> $GITHUB_OUTPUT
```

**Comment by alexevanczuk** [comment]
> Thanks for the PR!
> 
> > The docs (https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files) say to use:
> > > echo "{name}={value}" >> $GITHUB_OUTPUT
> 
> That would mean the updated lines should look like this, right?
> 
> ```shell
> echo "gem_version=$gem_version" >> $GITHUB_OUTPUT
> ```

Yes that seems right @CvX . I'm not sure what I was thinking with the original change! I've updated the PR.

**Review by cvx — APPROVED** [review]
Thanks! 😃

**Pull request merged**